### PR TITLE
Autogenerate fbs include files and rely on (refactored) include hierarchy

### PR DESCRIPTION
**Pull request by Wumpf**
### What

As implied by the title there's two things going on here:
* a fbs code generation step that generated include files
    * yes. it happend. we're generating code generators. 😱 
* a refactor of how we include things in general
   * previously, we sometimes did and sometimes did not include as-if we want to compile individual type definitions standalone. Practically, we never do that and things we don't do break. Degrading includes within type definitions to an annoying nuissance.
   * now, we almost never include in actual type definitions, leaning in on the fact that we only ever compile the new `entry_point.fbs`

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6923?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6923?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/6923)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by Wumpf** [comment]
hmm strange. why is it reporting codegen would do changes. doesn't change anything on my Windows machine.
It's not CRLF, must be some other similar sublety

File: crates/build/re_types_builder/src/codegen/fbs/mod.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        Ok(read_dir) => read_dir,
+        Err(err) => {
+            reporter.error_file(dir_path, err);
+            return contents;
+        }
+    };
+
+    let dir_name = dir_path.file_name().unwrap();
+
+    for entry in read_dir {
```
Comment (abey79) [inline crates/build/re_types_builder/src/codegen/fbs/mod.rs:71]: The order you get here might be OS dependent? Could that be the cause of the error?

**Review by abey79 — APPROVED** [review]
Great change! Also lgtm, but not super expert on the fbs so maybe another set of eyes?

File: crates/build/re_types_builder/src/codegen/fbs/mod.rs (hunk 1 above)
Comment (Wumpf) [reply crates/build/re_types_builder/src/codegen/fbs/mod.rs:71]: 👍 let's enforce alphabetical ordering

Comment (Wumpf) [reply crates/build/re_types_builder/src/codegen/fbs/mod.rs:72]: that was it, nice catch!

**Pull request merged**